src/SFA.DAS.Commitments.Api.IntegrationTests/DatabaseSetup/Generators/ApprenticeshipUpdateGenerator.cs

```diff
@@ -0,0 +1,31 @@
+﻿using System.Collections.Generic;
+using System.Linq;
+using System.Threading.Tasks;
+using AutoFixture;
+using SFA.DAS.Commitments.Api.IntegrationTests.DatabaseSetup.Entities;
+using SFA.DAS.Commitments.Api.IntegrationTests.Tests;
+
+namespace SFA.DAS.Commitments.Api.IntegrationTests.DatabaseSetup.Generators
+{
+    public class ApprenticeshipUpdateGenerator : Generator
+    {
+        public async Task<IEnumerable<DbSetupApprenticeshipUpdate>> Generate(int apprenticeshipsGenerated, long firstNewApprenticeshipId)
+        {
+            int apprenticeshipUpdatesToGenerate = (int)(apprenticeshipsGenerated * TestDataVolume.ApprenticeshipUpdatesToApprenticeshipsRatio);
```

Comment:
if `apprenticeshipsGenerated * TestDataVolume.ApprenticeshipUpdatesToApprenticeshipsRatio` is larger than int.MaxValue, this is going to throw an exception. Could this be changed to a long?

Reply:
Unless we explicitly compile with /checked or use the checked keyword it'll just silently overflow, which is actually worse!
We're actually unlikely to hit the issue on apprenticeshipUpdatesToGenerate, as it's only a small percentage of the number of Apprenticeship's we generate. We'd be more likely to hit it on the number of apprenticeships & data lock statuses we generate.
We could change all the counts of how many rows to generate to long, but it'd only be an issue if we wanted to generate more than 2,147,483,647 rows, which I suggest we're unlikely to want to do any time soon.